test/branded_token/constructor.js

```diff
@@ -51,8 +67,50 @@ contract('BrandedToken::constructor', async () => {
       );
     });
 
-    it('Reverts if conversionRate is zero', async () => {
+    it('Reverts if valueToken does not have an EIP20 decimals function', async () => {
       const valueToken = accountProvider.get();
+      const conversionRate = 35;
+      const conversionRateDecimals = 1;
+
+      await utils.expectRevert(
+        BrandedToken.new(
+          valueToken,
+          symbol,
+          name,
+          decimals,
+          conversionRate,
+          conversionRateDecimals,
+          organization,
+        ),
+        'Should revert as valueToken does not have an EIP20 decimals function.',
+        'ValueToken does not have an EIP20 decimals function.',
+      );
+    });
+
+    it('Reverts if valueToken decimals does not equal brandedToken decimals', async () => {
+      const valueToken = await deployValueToken(decimals + 1);
+
+      const conversionRate = 35;
+      const conversionRateDecimals = 1;
+
+      await utils.expectRevert(
+        BrandedToken.new(
+          valueToken,
+          symbol,
+          name,
+          decimals,
+          conversionRate,
+          conversionRateDecimals,
+          organization,
+        ),
+        'Should revert as valueToken decimals does not equal brandedToken decimals.',
+        'ValueToken decimals does not equal brandedToken decimals.',
+      );
+    });
+
+    it('Reverts if conversionRate is zero', async () => {
+      const valueToken = await deployValueToken(decimals);
```

Comment:
@abhayks1: I had considered this, but decided against it because this is not used in many of the tests and thought doing that and then adjusting in the places where it is not needed would be more code/work and hurt readability.

Thoughts?

Reply:
Ok we can keep the current implementation.